### PR TITLE
[TextFields] New styles: updated default, box, area

### PR DESCRIPTION
**Pull request by willlarche**

### CODE REVIEW

File: components/TextFields/examples/TextFieldBoxExample.swift [hunk 1]
```diff
@@ -0,0 +1,365 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
```
Comment (ianegordon) [inline components/TextFields/examples/TextFieldBoxExample.swift]: 2016 or 2017?

**Review by ianegordon — CHANGES_REQUESTED** [review]
Please verify copyright date for new files.

File: components/TextFields/src/MDCMultilineTextField.m [hunk 2]
```diff
@@ -513,16 +500,27 @@ - (void)setAttributedText:(NSAttributedString *)attributedText {
...
-- (UIButton *)clearButton {
-  return self.fundament.clearButton;
+- (UIBezierPath *)borderPath {
+  return self.fundament.borderPath ? self.fundament.borderPath : [self defaultBorderPath];
 }
 
-- (UIColor *)clearButtonColor {
-  return self.fundament.clearButtonColor;
+- (void)setBorderPath:(UIBezierPath *)borderPath {
+  if (self.fundament.borderPath != borderPath) {
```
Comment (ianegordon) [inline components/TextFields/src/MDCMultilineTextField.m]: != or isEqual: ?

File: components/TextFields/src/MDCTextField.m [hunk 3]
```diff
@@ -189,18 +192,44 @@ - (void)updateUnderlinePosition {
...
-- (UIButton *)clearButton {
-  return _fundament.clearButton;
+- (UIBezierPath *)borderPath {
+  return self.fundament.borderPath ? self.fundament.borderPath : [self defaultBorderPath];
 }
 
-- (UIColor *)clearButtonColor {
-  return _fundament.clearButtonColor;
+- (void)setBorderPath:(UIBezierPath *)borderPath {
+  if (self.fundament.borderPath != borderPath) {
```
Comment (ianegordon) [inline components/TextFields/src/MDCTextField.m]: =! or isEqual: ?

File: components/TextFields/src/MDCTextInputBorderView.h [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+#import <UIKit/UIKit.h>
+
+@interface MDCTextInputBorderView : UIView <NSCopying>
+
+/**
+ The color of the area inside the border.
+
+ Default is clear.
+ */
+@property(nonatomic, nullable, strong) UIColor *borderFillColor UI_APPEARANCE_SELECTOR;
```
Comment (ianegordon) [inline components/TextFields/src/MDCTextInputBorderView.h:26]: nullable or null_resettable?
If null_resettable, happy to open an issue and change in a later PR.

**Review by ianegordon — CHANGES_REQUESTED** [review]

File: components/TextFields/examples/TextFieldBoxExample.swift (hunk 1 above)
Comment (willlarche) [reply components/TextFields/examples/TextFieldBoxExample.swift]: Done.

File: components/TextFields/src/MDCTextInputBorderView.h (hunk 4 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputBorderView.h:26]: All UI_APPEARANCE_SELECTOR properties have to be nullable :(

**Review by ianegordon — APPROVED** [review]

File: components/TextFields/src/MDCTextField.m (hunk 3 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextField.m]: Ah. Done.

**Pull request merged**